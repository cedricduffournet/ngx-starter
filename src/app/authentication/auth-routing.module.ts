import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent
} from '@app/authentication/containers';
import { authConfig } from '@app/authentication/auth.config';

const AUTHENTICATION_ROUTES: Routes = [
  { path: 'login', component: LoginViewComponent },
  { path: 'resetting', component: SendResetPasswordViewComponent },
  {
    path: `${authConfig.path.passwordResetConfirm}/:token`,
    component: ResetPasswordViewComponent
  },
  { path: 'register', component: RegisterViewComponent },
  {
    path: `${authConfig.path.registerConfirm}/:token`,
    component: RegisterConfirmViewComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(AUTHENTICATION_ROUTES)],
  exports: [RouterModule]
})
export class AuthRoutingModule {}
