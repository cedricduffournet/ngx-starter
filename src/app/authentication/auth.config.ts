export const authConfig = {
  path: {
    registerConfirm: 'register/confirm',
    passwordResetConfirm: 'password/reset'
  },
  storageKey: 'authentication'
};
