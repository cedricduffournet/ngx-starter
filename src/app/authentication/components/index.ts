export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
