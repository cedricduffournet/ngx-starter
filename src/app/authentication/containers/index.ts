export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
