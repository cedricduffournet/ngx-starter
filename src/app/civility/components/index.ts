export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
