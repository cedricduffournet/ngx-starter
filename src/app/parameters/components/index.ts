export * from './referenced-table-menu/referenced-table-menu.component';
