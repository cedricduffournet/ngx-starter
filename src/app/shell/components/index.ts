export * from './navbar/navbar.component';
export * from './dropdown-navbar/dropdown-navbar.component';
export * from './menu/menu.component';
