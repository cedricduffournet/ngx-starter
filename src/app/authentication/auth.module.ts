import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityModule } from '@app/civility/civility.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { LogoModule } from '@app/shared/logo/logo.module';
import { AuthRoutingModule } from '@app/authentication/auth-routing.module';
import { AuthEffects } from '@app/authentication/state/effects';
import { reducers } from '@app/authentication/state/reducers';
import { AuthFacade } from '@app/authentication/state/auth.facade';

import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent,
  LogoutModalComponent
} from '@app/authentication/containers';

import {
  AuthLayoutComponent,
  LoginFormComponent,
  ResetPasswordFormComponent,
  SendResetPasswordFormComponent,
  RegisterFormComponent
} from '@app/authentication/components';

@NgModule({
  imports: [
    AuthRoutingModule,
    SharedModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
    LangSwitcherModule,
    LogoModule,
    CivilityModule
  ],
  declarations: [
    AuthLayoutComponent,
    LoginViewComponent,
    SendResetPasswordViewComponent,
    ResetPasswordViewComponent,
    RegisterViewComponent,
    RegisterConfirmViewComponent,
    LoginFormComponent,
    ResetPasswordFormComponent,
    SendResetPasswordFormComponent,
    LogoutModalComponent,
    RegisterFormComponent
  ],
  entryComponents: [LogoutModalComponent],
  providers: [AuthFacade]
})
export class AuthModule {}
