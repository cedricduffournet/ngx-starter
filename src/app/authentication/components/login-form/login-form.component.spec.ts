import { TestBed, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { LoginFormComponent } from '@app/authentication/components';

describe('LoginForm', () => {
  let fixture: ComponentFixture<LoginFormComponent>;
  let component: LoginFormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        ReactiveFormsModule,
        TranslateModule.forRoot(),
        FormModule
      ],
      declarations: [LoginFormComponent]
    });
    fixture = TestBed.createComponent(LoginFormComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should not emit event if email is invalid', () => {
    const credentials = {
      login: 'user',
      password: 'pass'
    };

    fixture.detectChanges();

    component.loginForm.setValue(credentials);

    spyOn(component.login, 'emit');
    component.onSubmit();

    expect(component.login.emit).not.toHaveBeenCalled();
  });

  it('should disable submit button form if processing', () => {
    component.processing = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit save event if the form is valid when submitted', () => {
    const credentials = {
      login: 'user@dasdas.fr',
      password: 'pass'
    };

    fixture.detectChanges();

    component.loginForm.setValue(credentials);

    spyOn(component.login, 'emit');
    component.onSubmit();

    expect(component.login.emit).toHaveBeenCalledWith(credentials);
  });

  it('should display an error message if provided', () => {
    component.error = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
