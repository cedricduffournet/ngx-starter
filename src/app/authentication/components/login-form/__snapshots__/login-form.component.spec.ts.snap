// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should be created 1`] = `
<app-login-form
  error="false"
  formBuilder={[Function FormBuilder]}
  login={[Function EventEmitter]}
  loginForm={[Function FormGroup]}
  processing="false"
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="login"
      ng-reflect-label="LOGIN"
      ng-reflect-name="login"
      ng-reflect-placeholder="PLACEHOLDER_LOGIN"
      ng-reflect-submitted="false"
      ng-reflect-type="email"
      type="email"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          LOGIN
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_LOGIN"
          type="email"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="password"
      ng-reflect-label="PASSWORD"
      ng-reflect-name="password"
      ng-reflect-placeholder="PLACEHOLDER_PASSWORD"
      ng-reflect-submitted="false"
      ng-reflect-type="password"
      type="password"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          PASSWORD
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_PASSWORD"
          type="password"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         CONNECT 
      </button>
    </div>
    <a
      class="btn btn-link btn-block"
      href="/resetting"
      ng-reflect-router-link="/resetting"
      ng-reflect-translate=""
      translate=""
    >
       RESET_PASSWORD 
    </a>
    <a
      class="btn btn-link btn-block"
      href="/register"
      ng-reflect-router-link="/register"
      ng-reflect-translate=""
      translate=""
    >
       REGISTER 
    </a>
  </form>
</app-login-form>
`;

exports[`LoginForm should disable submit button form if processing 1`] = `
<app-login-form
  error="false"
  formBuilder={[Function FormBuilder]}
  login={[Function EventEmitter]}
  loginForm={[Function FormGroup]}
  processing={[Function Boolean]}
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="login"
      ng-reflect-label="LOGIN"
      ng-reflect-name="login"
      ng-reflect-placeholder="PLACEHOLDER_LOGIN"
      ng-reflect-submitted="false"
      ng-reflect-type="email"
      type="email"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          LOGIN
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_LOGIN"
          type="email"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="password"
      ng-reflect-label="PASSWORD"
      ng-reflect-name="password"
      ng-reflect-placeholder="PLACEHOLDER_PASSWORD"
      ng-reflect-submitted="false"
      ng-reflect-type="password"
      type="password"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          PASSWORD
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_PASSWORD"
          type="password"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        disabled=""
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         CONNECT 
      </button>
    </div>
    <a
      class="btn btn-link btn-block"
      href="/resetting"
      ng-reflect-router-link="/resetting"
      ng-reflect-translate=""
      translate=""
    >
       RESET_PASSWORD 
    </a>
    <a
      class="btn btn-link btn-block"
      href="/register"
      ng-reflect-router-link="/register"
      ng-reflect-translate=""
      translate=""
    >
       REGISTER 
    </a>
  </form>
</app-login-form>
`;

exports[`LoginForm should display an error message if provided 1`] = `
<app-login-form
  error={[Function Boolean]}
  formBuilder={[Function FormBuilder]}
  login={[Function EventEmitter]}
  loginForm={[Function FormGroup]}
  processing="false"
  submitted="false"
>
  <div
    class="alert alert-danger"
    ng-reflect-translate=""
    translate=""
  >
     AUTHENTICATION_FAILED
  
  </div><form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="login"
      ng-reflect-label="LOGIN"
      ng-reflect-name="login"
      ng-reflect-placeholder="PLACEHOLDER_LOGIN"
      ng-reflect-submitted="false"
      ng-reflect-type="email"
      type="email"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          LOGIN
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_LOGIN"
          type="email"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="password"
      ng-reflect-label="PASSWORD"
      ng-reflect-name="password"
      ng-reflect-placeholder="PLACEHOLDER_PASSWORD"
      ng-reflect-submitted="false"
      ng-reflect-type="password"
      type="password"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          PASSWORD
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_PASSWORD"
          type="password"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         CONNECT 
      </button>
    </div>
    <a
      class="btn btn-link btn-block"
      href="/resetting"
      ng-reflect-router-link="/resetting"
      ng-reflect-translate=""
      translate=""
    >
       RESET_PASSWORD 
    </a>
    <a
      class="btn btn-link btn-block"
      href="/register"
      ng-reflect-router-link="/register"
      ng-reflect-translate=""
      translate=""
    >
       REGISTER 
    </a>
  </form>
</app-login-form>
`;
